setuptools/command/editable_wheel.py

```diff
@@ -466,8 +466,14 @@ def _create_file(self, relative_output: str, src_file: str, link=None):
     def _create_links(self, outputs, output_mapping):
         self.auxiliary_dir.mkdir(parents=True, exist_ok=True)
         link_type = "sym" if _can_symlink_files(self.auxiliary_dir) else "hard"
-        mappings = {self._normalize_output(k): v for k, v in output_mapping.items()}
-        mappings.pop(None, None)  # remove files that are not relative to build_lib
+        mappings = {
+            k: v
+            for k, v in (
+                (self._normalize_output(k), v) for k, v in output_mapping.items()
+            )
+            # remove files that are not relative to build_lib
+            if k is not None
+        }
```

Comment:
Probably a bit more readable... The name of the intermediary variable may be different, just used the first thing that came to my mind.

```suggestion
        normalised = ((self._normalize_output(k), v) for k, v in output_mapping.items())
        # remove files that are not relative to build_lib
        mappings = {k: v for k, v in normalised if k is not None}
```

Reply:
Looks great thanks